cli/tools/lint.rs

```diff
@@ -98,6 +105,85 @@ pub async fn lint_files(
   };
   let reporter_lock = Arc::new(Mutex::new(create_reporter(reporter_kind)));
 
+  let resolver = |changed: Option<Vec<PathBuf>>| {
+    let files_changed = changed.is_some();
+    let result =
+      collect_files(&args.clone(), &ignore.clone(), is_supported_ext).map(
+        |files| {
+          if let Some(paths) = changed {
+            files
+              .into_iter()
+              .filter(|path| paths.contains(path))
+              .collect::<Vec<_>>()
+          } else {
+            files
+          }
+        },
+      );
+    let paths_to_watch = args.clone();
+    async move {
+      if (files_changed || !watch)
+        && matches!(result, Ok(ref files) if files.is_empty())
+      {
+        ResolutionResult::Ignore
+      } else {
+        ResolutionResult::Restart {
+          paths_to_watch,
+          result,
+        }
+      }
+    }
+  };
+
+  let operation = |paths: Vec<PathBuf>| async {
+    let target_files_len = paths.len();
+    run_parallelized(paths, {
+      let reporter_lock = reporter_lock.clone();
+      let has_error = has_error.clone();
+      let lint_rules = lint_rules.clone();
+      move |file_path| {
+        let r = lint_file(file_path.clone(), lint_rules.clone());
+        handle_lint_result(
+          &file_path.to_string_lossy(),
+          r,
+          reporter_lock,
+          has_error,
+        );
+        Ok(())
+      }
+    })
+    .await?;
+    reporter_lock.lock().unwrap().close(target_files_len);
+    reporter_lock.lock().unwrap().reset_counter();
+
+    Ok(())
+  };
+
+  if watch {
+    file_watcher::watch_func(resolver, operation, "Lint").await?;
+  } else {
+    lint_files(
+      include_files.clone(),
+      exclude_files.clone(),
+      args.clone(),
+      reporter_lock.clone(),
+      lint_rules.clone(),
+      has_error.clone(),
+    )
+    .await?;
```

Comment:
It looks like `lint_files` is superfluous in this case and could be removed. Take a look at `cli/tools/fmt.rs` in `format` function - it calls `resolver()` and `operation()` manually to avoid code duplication